dll/win32/browseui/internettoolbar.cpp

```diff
@@ -907,33 +923,17 @@ HRESULT STDMETHODCALLTYPE CInternetToolbar::ShowDW(BOOL fShow)
             return hResult;
     }
 
-#if 0 // Why should showing the IDockingWindow change all bands? Related to CORE-17236
-    if (fMenuBar)
-    {
-        hResult = IUnknown_ShowDW(fMenuBar, fShow);
-        if (FAILED_UNEXPECTEDLY(hResult))
-            return hResult;
-    }
-
-    if (fControlsBar)
+    // TODO: Why should showing the IDockingWindow change all bands? Related to CORE-17236
+    int id;
+    IUnknown *pUnk;
+    for (UINT i = 0; SUCCEEDED(EnumBands(i, &id, &pUnk)); ++i)
     {
-        hResult = IUnknown_ShowDW(fControlsBar, fShow);
-        if (FAILED_UNEXPECTEDLY(hResult))
-            return hResult;
+        if (!pUnk)
+            continue;
+        BOOL visible = fShow && IsBandVisible(id) != S_FALSE;
```

Comment:
Since the IsBandVisible returns a HRESULT (you explained to me in a past PR that this was made also in such a way that it could return also another error besides S_OK/TRUE and S_FALSE, for example in case the band doesn't exist),
wouldn't this != S_FALSE check return TRUE also if the band does _NOT_ exist and the function returns another error?
Is this behaviour wanted then?

Reply:
Ideally it would check against S_OK but I don't want to risk breaking the menu again so it is very conservative and just treats anything other than "confirmed hidden" as visible.

Reply:
thanks